rules.d/99-systemd.rules.in

```diff
@@ -18,19 +18,14 @@ SUBSYSTEM=="ubi", TAG+="systemd"
 
 SUBSYSTEM=="block", TAG+="systemd"
 
-# We can't make any conclusions about suspended DM devices so let's just import previous SYSTEMD_READY state and skip other rules
-SUBSYSTEM=="block", ENV{DM_SUSPENDED}=="1", IMPORT{db}="SYSTEMD_READY"
-SUBSYSTEM=="block", ENV{DM_SUSPENDED}=="1", GOTO="systemd_end"
-
-SUBSYSTEM=="block", ACTION=="add", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", ENV{SYSTEMD_READY}="0"
+# DM_UDEV_DISABLE_OTHER_RULES_FLAG==1 means that the device shouldn't be probed.
+# Import previous SYSTEMD_READY state.
+SUBSYSTEM=="block", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", ENV{SYSTEMD_READY}=="", IMPORT{db}="SYSTEMD_READY"
 
 # Ignore encrypted devices with no identified superblock on it, since
 # we are probably still calling mke2fs or mkswap on it.
 SUBSYSTEM=="block", ENV{DM_UUID}=="CRYPT-*", ENV{ID_PART_TABLE_TYPE}=="", ENV{ID_FS_USAGE}=="", ENV{SYSTEMD_READY}="0"
 
-# Explicitly set SYSTEMD_READY=1 for DM devices that don't have it set yet, so that we always have something to import above
-SUBSYSTEM=="block", ENV{DM_UUID}=="?*", ENV{SYSTEMD_READY}=="", ENV{SYSTEMD_READY}="1"
```

Comment:
Then, when the device get `SYSTEMD_READY=1`?

Reply:
> Then, when the device get `SYSTEMD_READY=1`?

Devices are ready by default, so `ENV{SYSTEMD_READY}=1` is really needed only to cancel previous `ENV{SYSTEMD_READY}==0`.

Reply:
Ah, yes. Indeed. You are right.

Reply:
Then, so the third commit is mostly do nothing, right?

Reply:
My commit message is misleading. systemd itself reads `SYSTEMD_READY` as a boolean. All values except those that it doesn't parse as "false" (i.e. `0`, `no`, `n`, `false`, `f`, or `off`) cause systemd to assume the device is "ready". This includes the case where `SYSTEMD_READY` is not set.

So indeed, as far as systemd is concerned, `ENV{SYSTEMD_READY}=="1"` is equivalent with the variable being unset, and my assertion that setting this variable to 1 may be harmful is false. I'll fix that.

For udev itself, there's a subtle difference, which @jiayi0118 pointed out in c1a2ada89708d6aeeada496712cb24a4a58e75cc: `IMPORT{db}` is a matching token which evaluates to `true` if and only if the environment variable is set in the db. Setting `ENV{SYSTEMD_READY}="1"` makes sure that future compound rules containing `IMPORT{db}="SYSTEMD_READY"` will be applied.

IMO this is a higly surprising feature of the udev syntax, and I'd generally recommend to avoid using `IMPORT{db}` in compound rules.